src/cdk/schematics/ng-update/devkit-file-system.ts

```diff
@@ -46,10 +46,13 @@ export class DevkitFileSystem extends FileSystem {
     // The devkit tree does not expose an API for checking whether a given
     // directory exists. It throws a specific error though if a directory
     // is being read as a file. We use that to check if a directory exists.
+    // Depending on execution context, the instanceof check might not work,
+    // in which case we fall back to asserting exception class name.
     try {
       return this._tree.get(fileOrDirPath) !== null;
     } catch (e) {
-      if (e instanceof PathIsDirectoryException) {
+      if (e instanceof PathIsDirectoryException ||
+          e?.constructor?.name === 'PathIsDirectoryException') {
```

Comment:
This feels a little hacky, but maybe we don't need the try/catch anymore? The comment says that, at the time of writing, there was no API to check whether a directory exists, but looking at the methods on `this._tree`, it has both `getDir` and `exists`.

Reply:
I was uncertain, if this was due to internal restrictions.
I will investigate if this could simply be changed to exists.